AmplifyPlugins/Logging/Sources/AWSCloudWatchLoggingPlugin/AWSCloudWatchLoggingSessionController.swift

```diff
@@ -0,0 +1,193 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import AWSPluginsCore
+import Amplify
+import Combine
+import Foundation
+import AWSCloudWatchLogs
+import AWSClientRuntime
+import Network
+
+/// Responsible for setting up and tearing-down log sessions for a given category/tag according to changes in
+/// user authentication sessions.
+///
+/// - Tag: CloudWatchLogSessionController
+final class AWSCloudWatchLoggingSessionController {
+    
+    let logGroupName: String
+    let region: String
+    
+    var logLevel: LogLevel {
+        didSet {
+            self.session?.logger.logLevel = logLevel
+        }
+    }
+    
+    var client: CloudWatchLogsClientProtocol?
+    private let localStoreMaxSizeInMB: Int
+    private let credentialsProvider: CredentialsProvider
+    private let authentication: AuthCategoryUserBehavior
+    private let category: String
+    private let namespace: String?
+    private var session: AWSCloudWatchLoggingSession?
+    private var consumer: LogBatchConsumer?
+    private var batchSubscription: AnyCancellable? { willSet { batchSubscription?.cancel() } }
+    private var authSubscription: AnyCancellable? { willSet { authSubscription?.cancel() } }
+    private let logFilter: AWSCloudWatchLoggingFilterBehavior
+    private let networkMonitor: LoggingNetworkMonitor
+    
+    private var userIdentifier: String? {
+        didSet {
+            if oldValue != userIdentifier {
+                userIdentifierDidChange()
+            }
+        }
+    }
+
+    /// - Tag: CloudWatchLogSessionController.init
+    init(credentialsProvider: CredentialsProvider,
+         authentication: AuthCategoryUserBehavior,
+         logFilter: AWSCloudWatchLoggingFilterBehavior,
+         category: String,
+         namespace: String?,
+         logLevel: LogLevel,
+         logGroupName: String,
+         region: String,
+         localStoreMaxSizeInMB: Int,
+         userIdentifier: String?,
+         networkMonitor: LoggingNetworkMonitor
+    ) {
+        self.credentialsProvider = credentialsProvider
+        self.authentication = authentication
+        self.logFilter = logFilter
+        self.category = category
+        self.namespace = namespace
+        self.logLevel = logLevel
+        self.logGroupName = logGroupName
+        self.region = region
+        self.localStoreMaxSizeInMB = localStoreMaxSizeInMB
+        self.userIdentifier = userIdentifier
+        self.networkMonitor = networkMonitor
+    }
+    
+    func enable() {
+        updateSession()
+        updateConsumer()
+        connectProducerAndConsumer()
+    }
+    
+    func disable() {
+        self.batchSubscription = nil
+        self.authSubscription = nil
+        self.session = nil
+        self.consumer = nil
+    }
+
+    private func updateConsumer() {
+        do {
```

Comment:
You can just do

```swift
consumer = try? createConsumer()
```

Reply:
Good catch, code has been updated to the simplified recommendaiton.